Edits to pgd_bench content (#4789)
EnterpriseDB/docs: product_docs/docs/pgd/5/testingandtuning.mdx

### pgd_bench

The Postgres benchmarking application
[`pgbench`](https://www.postgresql.org/docs/current/pgbench.html) was

> extended in PGD 5.0 in the form of a new applications: pgd_bench.

Reviewer: ```suggestion
extended in PGD 5.0 in the form of a new application: pgd_bench.
```

[pgd_bench](/pgd/latest/reference/testingandtuning#pgd_bench) is a regular command-line utility that's added to the PostgreSQL bin
directory. The utility is based on the PostgreSQL pgbench tool but
supports benchmarking CAMO transactions and PGD-specific workloads.

Functionality of pgd_bench is a superset of those of pgbench but
requires the BDR extension to be installed to work properly.

Key differences include:

-   Adjustments to the initialization (`-i` flag) with the standard 
    pgbench scenario to prevent global lock timeouts in certain cases.
-   `VACUUM` command in the standard scenario is executed on all nodes.
-   pgd_bench releases are tied to the releases of the BDR extension

>     and are built against the corresponding PostgreSQL flavor. This is

Reviewer 2: Postgres distribution (add link to the choosing your distribution page)

>     and are built against the corresponding PostgreSQL flavor. This is

Reviewer: ```suggestion
    and are built against the corresponding Postgres distribution. This is
```

### Notes on pgd_bench usage

-   When using custom init-scripts, it's important to understand implications behind the DDL commands.
We generally recommend waiting for the secondary nodes to catch up on the data-load steps
before proceeding with DDL operations such as `CREATE INDEX`. The latter acquire global locks that
can't be acquired until the data load is complete and thus might time out.

-   No extra steps are taken to suppress client messages, such as `NOTICE` and `WARNING` messages emitted
by PostgreSQL and or any possible extensions, including the BDR extension. It's your
responsibility to suppress them by setting appropriate variables, such as `client_min_messages`, 
`bdr.camo_enable_client_warnings`, and so on.

## Performance testing and tuning

> PGD allows you to issue write transactions onto multiple master nodes. Bringing

Reviewer 2: master --> leader?

> PGD allows you to issue write transactions onto multiple master nodes. Bringing

Reviewer: ```suggestion
PGD allows you to issue write transactions onto multiple leader nodes. Bringing
```

those writes back together onto each node has a performance cost.

> First, replaying changes from another node has a CPU cost an an I/O cost,

Reviewer 2: the original was correct - it's a list of 3 things:
CPU Cost
I/O Cost
Generates WAL records

could say has CPU and I/O cost and generates WAL records

> First, replaying changes from another node has a CPU cost an an I/O cost,

Reviewer: ```suggestion
First, replaying changes from another node has a CPU cost, an I/O cost,
```

and it generates WAL records. The resource use is usually less

> than in the original transaction since CPU overheads are lower as a result

Reviewer 2: CPU overhead is lower

> than in the original transaction since CPU overheads are lower as a result

Reviewer: ```suggestion
than in the original transaction since CPU overhead is lower as a result
```

### Notes on pgd_bench usage

-   When using custom init-scripts, it's important to understand implications behind the DDL commands.
We generally recommend waiting for the secondary nodes to catch up on the data-load steps
before proceeding with DDL operations such as `CREATE INDEX`. The latter acquire global locks that
can't be acquired until the data load is complete and thus might time out.

-   No extra steps are taken to suppress client messages, such as `NOTICE` and `WARNING` messages emitted
by PostgreSQL and or any possible extensions, including the BDR extension. It's your
responsibility to suppress them by setting appropriate variables, such as `client_min_messages`, 
`bdr.camo_enable_client_warnings`, and so on.

## Performance testing and tuning

> PGD allows you to issue write transactions onto multiple leader nodes. Bringing

Reviewer: ```suggestion
PGD allows you to issue write transactions onto multiple nodes. Bringing
```

master has a meaning in a multi-master environment. Leader is singular though. Simpler to omit master and leader.
